core/src/main/java/fr/sncf/osrd/standalone_sim/SafetySpeed.kt

```diff
@@ -0,0 +1,94 @@
+package fr.sncf.osrd.standalone_sim
+
+import fr.sncf.osrd.api.api_v2.standalone_sim.SimulationScheduleItem
+import fr.sncf.osrd.conflicts.TravelledPath
+import fr.sncf.osrd.railjson.schema.schedule.RJSTrainStop
+import fr.sncf.osrd.sim_infra.api.Path
+import fr.sncf.osrd.sim_infra.api.RawInfra
+import fr.sncf.osrd.sim_infra.api.Route
+import fr.sncf.osrd.sim_infra.api.ZonePath
+import fr.sncf.osrd.sim_infra.api.ZonePathId
+import fr.sncf.osrd.sim_infra.impl.ChunkPath
+import fr.sncf.osrd.utils.DistanceRangeMap
+import fr.sncf.osrd.utils.distanceRangeMapOf
+import fr.sncf.osrd.utils.indexing.StaticIdx
+import fr.sncf.osrd.utils.indexing.StaticIdxList
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.Speed
+import fr.sncf.osrd.utils.units.kilometersPerHour
+import fr.sncf.osrd.utils.units.meters
+
+/**
+ * Compute safety speed ranges, areas where the train has a lower speed limit because of a scheduled
+ * stop. For details, see https://osrd.fr/en/docs/reference/design-docs/timetable/#modifiable-fields
+ * (or google "VISA (VItesse Sécuritaire d'Approche)" for resources in French)
+ */
+fun makeSafetySpeedRanges(
+    infra: RawInfra,
+    chunkPath: ChunkPath,
+    routes: StaticIdxList<Route>,
+    schedule: List<SimulationScheduleItem>
+): DistanceRangeMap<Speed> {
+    val zonePaths = routes.flatMap { infra.getRoutePath(it) }
+    val zonePathStartOffset = getRoutePathStartOffset(infra, chunkPath, zonePaths)
+    val signalOffsets = getSignalOffsets(infra, zonePaths, zonePathStartOffset)
+
+    val stopsWithSafetySpeed =
+        schedule.filter {
+            it.stopFor != null && it.receptionSignal != RJSTrainStop.RJSReceptionSignal.OPEN
+        }
+
+    val res = distanceRangeMapOf<Speed>()
+    for (stop in stopsWithSafetySpeed) {
+        val nextSignalOffset =
+            signalOffsets.firstOrNull { it >= stop.pathOffset }?.distance ?: chunkPath.length
+        res.put(
```

Comment:
If we get the signalSightDistance, we can add a condition before all that, no?
Not sure if we should check the type of signal here.

Reply:
I'm leaving this unresolved until we talk with @axrolld, I don't know the expected behavor there. Apparently there *can* be stops on closed signals even if it's not seen, but I'm not sure what the expected behavior would be.

Reply:
To be confirmed, after discussion with @Yohh and @Uriel-Sautron:
- [x] it looks like there is no realistic counter-example for ignoring signal sight (at least for 🇫🇷 ).
The 200m safety speed of approach is supposed to be respected even when the sight distance is less than 200m (cases of turns, fog, ...)
Cases of stops out of the sight of signal are also most probably security stops, not planned, and the behavior in those case is even more restrictive: safety speed all the way to the next block-signal.

- [ ] As for the type, it looks like the type of the signal where the safety speed of approach must be applied is the next signal that is delimiting a block, so this should be handled when confirmed.